### PR TITLE
MSVC native support

### PR DESCRIPTION
**Pull request by gcask**
Native support for MSVC (2017 toolchain, matching SFML).

Currently focusing on 32b only.

Most of was just conformance (now that MS is doing a much better job at it, it's a lot easier).

CMake install targets would need an update (doesn't need all the DLLs from MinGW, requires the SFML ones to be copied over).

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -46,8 +46,12 @@ if(CMAKE_C_COMPILER_ID STREQUAL "GNU")
     endif()
 endif()
 
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Werror=return-type")
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Werror=return-type")
+if(MSVC)
+	add_definitions(-DNOMINMAX)
```
Comment (daid) [inline CMakeLists.txt:50]: maybe MSVC needs some warning flags as well? I'm not familiar enough with MSVC to judge that.

File: src/main.cpp [hunk 2]
```diff
@@ -157,7 +157,7 @@ int main(int argc, char** argv)
     new DirectoryResourceProvider("resources/");
     new DirectoryResourceProvider("scripts/");
     new DirectoryResourceProvider("packs/SolCommand/");
-    PackResourceProvider::addPackResourcesForDirectory("packs");
+    PackResourceProvider::addPackResourcesForDirectory("packs/");
```
Comment (daid) [inline src/main.cpp:160]: As pointed out in the SeriousProton pull request, I don't like the API change.

**Review by daid — CHANGES_REQUESTED** [review]
Other then the API change, it's good.

File: CMakeLists.txt (hunk 1 above)
Comment (gcask) [reply CMakeLists.txt:50]: There are several changes that needs to happen yeah - I intend to do a pass on CMake config for msvc later on.

File: src/main.cpp (hunk 2 above)
Comment (gcask) [reply src/main.cpp:160]: I relaxed the API change.
However, I would argue this is worth keeping for the sake of consistency?
All other paths (besides mod) are currently forward slash terminated...

I'll revert this if it's really a bother though :)

Comment (daid) [reply src/main.cpp:160]: Oh, don't mind it here, only mind the API change :)

File: src/packResourceProvider.cpp [hunk 3]
```diff
@@ -67,6 +69,25 @@ std::vector<string> PackResourceProvider::findResources(const string searchPatte
 
 void PackResourceProvider::addPackResourcesForDirectory(const string directory)
 {
+#ifdef _WIN32
+    WIN32_FIND_DATAA data;
+    assert(directory.endswith("/"));
```
Comment (daid) [inline src/packResourceProvider.cpp:74]: Hench, this one...

Comment (gcask) [reply src/packResourceProvider.cpp]: 🤦 I thought both were in Serious Proton (and thought I dreamed the second one since I could no longer find it 😂 )

Comment (daid) [reply src/packResourceProvider.cpp]: Never said the code is well organized ;-)

Comment (gcask) [reply src/packResourceProvider.cpp]: It's not *so* bad actually.

**Pull request merged**